openslides/config/views.py

```diff
@@ -41,17 +61,25 @@ def update(self, request, *args, **kwargs):
         if not request.user.has_perm('config.can_manage'):
             self.permission_denied(request)
 
-        # Check if pk is a valid config variable key.
         key = kwargs['pk']
-        if key not in config:
+        value = request.data['value']
+
+        # Check if pk is a valid config variable key.
+        try:
+            config_variable = config.get_config_variables()[key]
+        except ConfigNotFound:
             raise Http404
 
-        # Validate value.
-        form_field = config.get_config_variables()[key].form_field
-        value = request.data['value']
-        if form_field:
+        # Validate datatype.
+        expected_type = CONFIG_HTML_INPUT_TYPE_MAPPING[config_variable.html_input_type]
+        if not isinstance(value, expected_type):
+            raise ValidationError({
+                'detail': _('Wrong datatype. Expected %s, got %s.') % (expected_type, type(value))})
+        if config_variable.html_input_type == 'radio' and not dict(config_variable.choices).get(value):
```

Comment:
Suggestion: Move the validation code into the config-api, inside the __setitem__ method, so that it is also called at other places.